webapp/channels/src/selectors/onboarding.ts

```diff
@@ -13,9 +13,9 @@ import {RecommendedNextStepsLegacy, Preferences} from 'utils/constants';
 
 import type {GlobalState} from 'types/store';
 
-const getCategory = makeGetCategory();
+const getABTestCategory = makeGetCategory();
```

Comment:
0/5, but maybe we should change `makeGetCategory` to take the category name as a parameter and just return a selector which only needs the state. Unlike other selector factories like `makeGetChannel`, callers aren't likely to be changing that parameter after the selector is created, and it makes some of these nicer to use

```ts
export const getABTestPreferences = makeGetCategory(Preferences.AB_TEST_PREFERENCE_VALUE);

const getNextStepsCategory = makeGetCategory(Preferences.RECOMMENDED_NEXT_STEPS);
const getOnboardingTaskCategory = makeGetCategory(Preferences.RECOMMENDED_NEXT_STEPS));

export const legacyNextStepsNotFinished = createSelector(
    'legacyNextStepsNotFinished',
    (state: GlobalState) => getNextStepsCategory(state),
    // ...
);
```

Reply:
I would say it's out of scope for this PR, would be quite a bit of refactoring to add on top of this. It does make sense to make the change though, I've created a ticket https://mattermost.atlassian.net/browse/MM-58111